pkg/cloudprovider/provider/digitalocean/provider.go

```diff
@@ -342,20 +360,34 @@ func (p *provider) Delete(machine *v1alpha1.Machine) error {
 	if err != nil {
 		return fmt.Errorf("failed to convert instance id %s to int: %v", i.ID(), err)
 	}
-	_, err = client.Droplets.Delete(ctx, doID)
-	return err
+
+	rsp, err := client.Droplets.Delete(ctx, doID)
+	if err != nil {
+		if terminalError := httpStatusToTerminalError(rsp.StatusCode); terminalError != nil {
```

Comment:
I slightly improved the way we return terminal errors in this pull
https://github.com/kubermatic/machine-controller/pull/136